src/libraries/System.Net.WebSockets/tests/WebSocketDeflateTests.cs

```diff
@@ -434,11 +434,11 @@ public async Task PayloadShouldHaveSimilarSizeWhenSplitIntoSegments(int windowBi
             int frameSize = 2 << windowBits;
 
             byte[] message = new byte[frameSize * 10];
-            Random random = new(0);
+            new Random(0).NextBytes(message);
 
             for (int i = 0; i < message.Length; ++i)
             {
-                message[i] = (byte)random.Next(maxValue: 10);
+                message[i] %= 10;
             }
 
             await client.SendAsync(message, WebSocketMessageType.Binary, true, CancellationToken);
```

Comment:
What about also using a less eager cancelation token since you suspect it's hanging. As a rule of thumb, I never have a test time out sooner than 30 seconds. I have seen tests that normally take a few 100ms end up taking 10 seconds on some hardware / with other tests running. 5 seconds is aggressive.

Reply:
I have put up a PR for this https://github.com/dotnet/runtime/pull/52086 since we need to stop it breaking others.

Reply:
@danmoseley thanks, I didn't have experience here, and I thought 5 seconds is a lot of time. Usually this test should run in about 1-10ms range. Why I think it's the random and not something else is that if you look at some of the failures, you will see timeouts around 7-10 seconds. However the cancellation token timeout is 5 seconds. This means that the code before the send takes that amount of time, and the only thing there is the random.

See the test results for this PR here: https://dev.azure.com/dnceng/public/_build/results?buildId=1113989&view=ms.vss-test-web.build-test-results-tab&runId=33947022&resultId=126924&paneView=debug. It takes now 131ms to execute for 15 bits.